### DISCUSSION TITLE
Integrated Terminal support

### DISCUSSION THREAD

**Discussion by tsxjsx** [scalameta/metals#6161] [General]
Metals support integrated terminal to the clients? client can implement integrated terminal by using metals language server?like eclipse jdtls support the integrated terminal

**Comment by tgodzik** [comment]
The terminal support depends on the editor itself. What kind of terminal would you find useful otherwise? Do you mean for sbt?

**Reply by tsxjsx** [reply]
thanks for replying tgodzik 
Despite the support for the runInTerminal capability in VSCode, Metals does not initiate runInTerminal requests to the client IDE, as evidenced by the absence of such requests in the trace logs (bsp.trace.json or dap-server.trace.json)
(vscode ide  support for integrated terminal)

**Reply by tgodzik** [reply]
It's specific to VS Code, there is no LSP equivalent, so this would most likely be handled in VS Code plugin itself

We use the terminal hovere when running without debugging, but not sure if it is what you have in mind. 

If you have a need of a specific terminal you already set it up in VS Code itself.

**Comment by tsxjsx** [comment]
which is debug adapter for metals ?

**Reply by tgodzik** [reply]
https://github.com/scalacenter/scala-debug-adapter
